packages/dds/tree/src/simple-tree/core/treeNodeSchema.ts

```diff
@@ -18,6 +18,13 @@ import type { InternalTreeNode, TreeNode, Unhydrated } from "./types.js";
  * Captures the schema both as runtime data and compile time type information.
  * Use {@link SchemaFactory} to define schema.
  * Use `Tree.schema(value)` to lookup the schema for a {@link TreeNode} or {@link TreeLeafValue}.
+ * @privateRemarks
+ * TODO:
+ * The long lists of type parameters here are awkward to deal with.
+ * Switching to (or adding an option to use)
+ * an interface based pattern with unordered named parameters for types like this would be a good idea.
+ * The related `@system` types should be able to simple be ported to the new pattern, but stable public one like this will need to support both:
```

Comment:
Typo here. 
```suggestion
 * The related `@system` types should be simple to port to the new pattern, but stable public one like this will need to support both:
```